pkg/services/profiles/profiles.go

```diff
@@ -34,3 +40,29 @@ func NewService(clientSet kubernetes.Interface, log logger.Logger) *ProfilesSvc
 		Logger:    log,
 	}
 }
+
+func (s *ProfilesSvc) discoverHelmRepository(ctx context.Context, opts GetOptions) (types.NamespacedName, string, error) {
+	availableProfile, version, err := s.GetProfile(ctx, opts)
+	if err != nil {
+		return types.NamespacedName{}, "", fmt.Errorf("failed to get profiles from cluster: %w", err)
+	}
+
+	return types.NamespacedName{
+		Name:      availableProfile.HelmRepository.Name,
+		Namespace: availableProfile.HelmRepository.Namespace,
+	}, version, nil
+}
+
+func getGitCommitFileContent(files []*gitprovider.CommitFile, filePath string) string {
+	for _, f := range files {
+		if f.Path != nil && *f.Path == filePath {
+			if f.Content == nil || *f.Content == "" {
```

Comment:
```suggestion
			if f.Content == nil || f.Content != nil && *f.Content == "" {
```

Reply:
isnt this implied? if `f.Content == nil` is false, that means `f.Content != nil`

Reply:
Yep, you're right. Nice. :)